### ISSUE TITLE
Unable to create donut pie chart

### ISSUE THREAD

**Issue by abhayathapa** [lumean/svg-graph2#6]
Is there any option to create donut pie chart like this `https://image-charts.com/chart?cht=pd&chtt=Blend&chs=978x300&chd=t:20,20,30,400,50&chl=20%C2%B0|20%C2%B0|30%C2%B0|40%C2%B0|50%C2%B0&chli=95K&chdl=Expansion|Payroll|Equipment` ?

**Comment by lumean** [comment]
Hi thanks for your request. Currently this type of graph is not supported. However I'm planning to integrate c3js in a next release (see http://c3js.org/examples.html). This will bring support, but I cannot commit on a timeline.

**Comment by abhayathapa** [comment]
Thanks @lumean

**Comment by abhayathapa** [comment]
@lumean Since, im planning on using this, can you give rough estimate on when this feature will be integrated. Thanks again

**Comment by lumean** [comment]
I can't promise anything since I'm just working on this in my free time. I plan to do something the next two weeks. 
Note: it will not be a native ruby integration. You will actually write a c3js java script with a bit of ruby string interpolation/ruby to json conversion. With the advantage that you can fully rely on the c3js documentation and write plain javascript code for your graphs.  You can copy paste 1:1 from the c3js examples and then put your data as follows:

```
my_data1 = ['data1', 30, 200, 100, 400, 150, 250] 
my_data2 = ['data2', 300, 20, 10, 40, 15, 25]
# using json notation for ruby hash 
graph_data = {
  bindto: '#this_is_my_awesom_graph',
  data: {
    columns: [ my_data1, my_data2],
...
}
gen1.javascript() {
  "var chart = c3.generate(#{JSON(graph_data)});"
}
```
Alternatively instead of using a ruby hash you could use a plain string containing javascript code:
```
my_data1 = ['data1', 30, 200, 100, 400, 150, 250] 
my_data2 = ['data2', 300, 20, 10, 40, 15, 25]
javascript_data = <<HEREDOC
graph_data = {
  bindto: '#this_is_my_awesom_graph',
  data: {
    columns: [ #{JSON(my_data1)}, #{JSON(my_data2)}],      <<<<<<< convert ruby to json.
    type: 'donut'
  }
}
setTimeout(function () {
    chart.unload({
        ids: 'data1'
    });
    chart.unload({
        ids: 'data2'
    });
}, 2500);
HEREDOC

gen1.javascript() {
  javascript_data
}
```


I have to think a bit about the API I will use for SVG::Graph. And what the output is. E.g. if the output is a html page I can easily include the c3js and d3js libraries in the `<head>` section, but if the output should just be an xml subtree with `<svg>` or `<div>` root node which will then basically anyway just contain a javascript child node, as  I don't think it makes sence to include references to the libraries but we should assume that a page which then includes that output should have the proper libraries included. The graph itself then is not anymor produced as 'static' svg but dynamically rendered by the browser/svgviewer from the javascript code.

What sort of output would you need for your project?  

See here for the complete example in another project I did:  https://github.com/lumean/xhtml-report-generator/blob/master/examples/graph.rb#L10-L55    resulting in  https://cdn.rawgit.com/lumean/xhtml-report-generator/master/examples/graph.xhtml

**Comment by lumean** [comment]
Released `2.2.0.beta` today.  Please have a look at the examples / test folder for code examples.
Use `gem install svg-graph --pre` to install the beta.
